### ISSUE TITLE
Fontmanager fails to build under Node.js v9 and 10

### ISSUE THREAD

**Issue by mririgoyen** [devongovett/font-manager#24]
When trying to install fontmanger under Node 9 or the newly released 10, fontmanager fails to build.

Here is the output of the failure on Node 10:

```
error /Users/mirigoyen/Git/projectname/node_modules/font-manager: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: /Users/mirigoyen/Git/projectname/node_modules/font-manager
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.2
gyp info using node@10.0.0 | darwin | x64
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn /usr/local/bin/python2
gyp info spawn args [ '/Users/mirigoyen/Git/projectname/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/mirigoyen/Git/projectname/node_modules/font-manager/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/mirigoyen/Git/projectname/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/mirigoyen/.node-gyp/10.0.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/mirigoyen/.node-gyp/10.0.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/mirigoyen/Git/projectname/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/mirigoyen/.node-gyp/10.0.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/mirigoyen/Git/projectname/node_modules/font-manager',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/fontmanager/src/FontManager.o
In file included from ../src/FontManager.cc:5:
In file included from ../node_modules/nan/nan.h:194:
../node_modules/nan/nan_maybe_43_inl.h:88:15: error: no member named 'ForceSet' in 'v8::Object'
  return obj->ForceSet(GetCurrentContext(), key, value, attribs);
         ~~~  ^
../node_modules/nan/nan_maybe_43_inl.h:220:17: error: no member named 'CloneElementAt' in 'v8::Array'
  return array->CloneElementAt(GetCurrentContext(), index);
         ~~~~~  ^
In file included from ../src/FontManager.cc:5:
In file included from ../node_modules/nan/nan.h:200:
In file included from ../node_modules/nan/nan_new.h:189:
../node_modules/nan/nan_implementation_12_inl.h:40:38: error: too few arguments to function call, expected 2, have 1
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
         ~~~~~~~~~~~~~~~~~~~~~~      ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/v8.h:4866:3: note: 'New' declared here
  static Local<Value> New(Isolate* isolate, bool value);
  ^
In file included from ../src/FontManager.cc:5:
In file included from ../node_modules/nan/nan.h:200:
In file included from ../node_modules/nan/nan_new.h:189:
../node_modules/nan/nan_implementation_12_inl.h:40:60: error: expected '(' for function-style cast or type construction
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                                          ~~~~~~~~~~~~~~~~~^
../node_modules/nan/nan_implementation_12_inl.h:40:62: error: expected expression
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                                                             ^
In file included from ../src/FontManager.cc:5:
../node_modules/nan/nan.h:820:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/FontManager.cc:5:
../node_modules/nan/nan.h:834:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:164:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/FontManager.cc:5:
../node_modules/nan/nan.h:848:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:157:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/FontManager.cc:5:
../node_modules/nan/nan.h:1461:31: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return scope.Escape(node::MakeCallback(
                              ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/FontManager.cc:5:
../node_modules/nan/nan.h:1957:7: error: no viable conversion from 'imp::NativeGetter' (aka 'void (*)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value> &)') to 'Local<v8::Name>'
    , getter_
      ^~~~~~~
/Users/mirigoyen/.node-gyp/10.0.0/include/node/v8.h:202:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'imp::NativeGetter' (aka 'void (*)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value> &)') to 'const v8::Local<v8::Name> &' for 1st argument
class Local {
      ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/v8.h:202:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'imp::NativeGetter' (aka 'void (*)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value> &)') to 'v8::Local<v8::Name> &&' for 1st argument
/Users/mirigoyen/.node-gyp/10.0.0/include/node/v8.h:206:13: note: candidate template ignored: could not match 'Local<type-parameter-0-0>' against 'void (*)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value> &)'
  V8_INLINE Local(Local<S> that)
            ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/v8.h:3215:43: note: passing argument to parameter 'name' here
      Local<Context> context, Local<Name> name,
                                          ^
/Users/mirigoyen/.node-gyp/10.0.0/include/node/v8.h:213:5: error: assigning to 'v8::Context *' from incompatible type 'v8::String *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/mirigoyen/.node-gyp/10.0.0/include/node/v8.h:167:39: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                      ^~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1956:7: note: in instantiation of function template specialization 'v8::Local<v8::Context>::Local<v8::String>' requested here
      name
      ^
4 warnings and 7 errors generated.
make: *** [Release/obj.target/fontmanager/src/FontManager.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/mirigoyen/Git/projectname/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Darwin 17.5.0
gyp ERR! command "/Users/mirigoyen/.nvm/versions/node/v10.0.0/bin/node" "/Users/mirigoyen/Git/projectname/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/mirigoyen/Git/projectname/node_modules/font-manager
```

**Comment by Eugeny** [comment]
I have published @moriyoshi's fix on NPM as `font-manager-nan-fix@0.2.2`

**Comment by devongovett** [comment]
hey sorry, I just published `font-manager@0.3.0` with that fix.
